dubbo-monitor/dubbo-monitor-api/src/main/java/org/apache/dubbo/monitor/support/AbstractMonitorFactory.java

```diff
@@ -79,10 +70,13 @@ public Monitor getMonitor(URL url) {
             }
 
             final URL monitorUrl = url;
-            final ListenableFutureTask<Monitor> listenableFutureTask = ListenableFutureTask.create(new MonitorCreator(monitorUrl));
-            listenableFutureTask.addListener(new MonitorListener(key));
-            executor.execute(listenableFutureTask);
-            FUTURES.put(key, listenableFutureTask);
+            final CompletableFuture<Monitor> completableFuture = CompletableFuture.supplyAsync(()->{
+                Monitor newMonitor = AbstractMonitorFactory.this.createMonitor(monitorUrl);
+                return newMonitor;
+
+            },executor);
+            completableFuture.thenRunAsync(new MonitorListener(key));

```

Comment:
We already defined an executor, specify this executor for `thenRunAsync ` may be better.